Add XML trace specification -- no code change

xbauch opened the pull request:
and a reference in the documentation.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

hannes-steffenhagen-diffblue commented on doc/cprover-manual/cbmc-tutorial.md:
````diff
@@ -130,6 +130,16 @@ access. If you add a branch to the example that requires that `argc>=3`,
+```
+
+The specification of the XML trace output can be found here:
+[XML Specification](https://raw.githubusercontent.com/diffblue/cbmc/develop/doc/assets/xml_spec.tex)
````
This should contain information on how the file is intended to be processed. I think this should work with most `LaTeX` backends with `-shell-escape`?

hannes-steffenhagen-diffblue replied:
Also, maybe even consider producing it as part of our documentation somehow? The LaTeX itself isn’t super human readable unprocessed.

thk123 replied:
Yeah :no_entry_sign: there must be either a script or a readme next to it explaining how to build it.

majakusber commented:
# [Codecov](https://codecov.io/gh/diffblue/cbmc/pull/5222?src=pr&el=h1) Report
> Merging [#5222](https://codecov.io/gh/diffblue/cbmc/pull/5222?src=pr&el=desc) into [develop](https://codecov.io/gh/diffblue/cbmc/commit/5b4bcc6048dd99fcd07d93836721d66a97b293ac&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[![Impacted file tree graph](https://codecov.io/gh/diffblue/cbmc/pull/5222/graphs/tree.svg?width=650&height=150&src=pr&token=WU7GQX8lAx)](https://codecov.io/gh/diffblue/cbmc/pull/5222?src=pr&el=tree)

```diff
@@           Coverage Diff            @@
##           develop    #5222   +/-   ##
========================================
  Coverage    67.43%   67.43%           
========================================
  Files         1157     1157           
  Lines        95351    95351           
========================================
  Hits         64303    64303           
  Misses       31048    31048           
```

| Flag | Coverage Δ | |
|---|---|---|
| #cproversmt2 | `42.67% <ø> (ø)` | :arrow_up: |
| #regression | `63.96% <ø> (ø)` | :arrow_up: |
| #unit | `31.95% <ø> (ø)` | :arrow_up: |


------

[Continue to review full report at Codecov](https://codecov.io/gh/diffblue/cbmc/pull/5222?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/diffblue/cbmc/pull/5222?src=pr&el=footer). Last update [5b4bcc6...e6c2933](https://codecov.io/gh/diffblue/cbmc/pull/5222?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

majakusber commented:
Why is my face on the codecov comment? :eyes:

xbauch commented:
> Why is my face on the codecov comment?

Wait. You didn't post the coverage report here?

majakusber commented:
Nope, never seen this PR, never did anything with codecov :woman_shrugging: I only learned about this because I got emails from github after Thomas added his comment (but not earlier).

martin-cs reviewed (approved):
I appreciate that documentation is often a thankless task so forgive me if I am making this harder but would it be possible to produce the spec as an XSD file or something so that we add a regression test that validates the output of CBMC against it.  That way when someone changes the format we should at least have a flag to update the documentation.

petr-bauch committed:
Add XML trace specification

and a reference in the documentation.

hannes-steffenhagen-diffblue committed:
Fix some typos/syntax errors in xml_spec.tex

Also replace some latex instructions that pandoc can’t handle with
some it can.

hannes-steffenhagen-diffblue committed:
Add XSD that describes output of cbmc --trace

Note that this is based on observed outputs from cbmc, there is currently no
mechanism in place that enforces CBMC to conform to this.

We have discussed adding a regression test pass for checking that at least our
own regression tests conform to the spec. In a future PR.

hannes-steffenhagen-diffblue commented:
Turning it off and turning it on again

thk123 commented on doc/cprover-manual/cbmc-tutorial.md:
```diff
@@ -130,6 +130,18 @@ access. If you add a branch to the example that requires that `argc>=3`,
+The specification of the XML trace output can be found here: [XML
+Specification](https://raw.githubusercontent.com/diffblue/cbmc/develop/doc/assets/xml_spec.tex)
+and can be build by `pdflatex -shell-escape xml_spec.tex`. Alternatively, you
+view it in Markdown [here](../../../assets/xml_spec).
```
This link doesn't seem to work, I think you only need to go up two directories
```suggestion
view it in Markdown [here](../../assets/xml_spec).
```

hannes-steffenhagen-diffblue replied:
The link doesn’t work because it links to the develop documentation, which doesn’t have this yet I think.

chrisr-diffblue reviewed (approved):
This looks like a great improvement.
